File: Lession_7/seminar/Shapes.py
from abc import ABC, abstractmethod
from math import pi
from typing import List

class Shape(ABC):
    @abstractmethod
    def get_area(self):
        pass

class SquareShape(Shape):
    def __init__(self,side):
        self.side = side

    def get_area(self):
        print('Площадь квадрата, со стороной: ',self.side, ' = ')
        return self.side * self.side

class CircleShape(Shape):
    def __init__(self,radius):
        self.radius = radius
    
    def get_area(self):
        print('Диаметр круга, с радиусом: ', self.radius, ' = ')
        return pi * self.radius * self.radius
    
shapes: List[Shape] = [SquareShape(2), CircleShape(4)]
# shapes =  [SquareShape(2), CircleShape(4)]

for shape in shapes:
    print(shape.get_area())

File: Lession_7/seminar/Shapes.java
package seminar;

import java.util.Arrays;

interface Shape {
    float getArea ();
}

class SquareSwhape implements Shape {
    private float side;

    public SquareSwhape(float side) {
        this.side = side;
    }

    @Override
    public float getArea() {
        return side * side;
    }

    
}

class CircleShape implements Shape {
    float radius;

    public CircleShape(float radius) {
        this.radius = radius;
    }

    @Override
    public float getArea() {

        return (float) Math.PI * radius * radius;
    }

    class Main {
        public static void main(String[] args) {
            Iterable<Shape> shapes = Arrays.asList(
                new SquareSwhape(1),
                new CircleShape(1)
            );

            for (Shape shape : shapes) {
                System.out.println(shape.getArea());
            }
        }
    }

    
}